Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -265,6 +265,11 @@ namespace ROS2
             auto* component = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(entity);
             AZ_Assert(component, "ROS2 Frame Component does not exist for %s", entityId.ToString().c_str());
             component->SetFrameID(AZStd::string(link->name.c_str(), link->name.size()));
+            // non-static transforms are applied to movable link and root.
+            if (!link->parent_joint || Utils::IsMovableLink(link))
```

Comment:
this would not be needed if we set it in ROS2Frame Activate instead and would be encapsulated.